### PR TITLE
Add stable Tensor overloads for index_select, floor_divide, and is_pinned (2.14+) 

### PR DESCRIPTION
**Pull request by cleonard530**
## Summary
- Add `torch::stable::{index_select,floor_divide,is_pinned}` Tensor–Tensor wrappers (gated at 2.14) via `torch_call_dispatcher`
- Document the new APIs in the stable operators docs
- Cover them in the 2.14 libtorch-agnostic extension tests

'stacked' on top of:
https://github.com/pytorch/pytorch/pull/192083
https://github.com/pytorch/pytorch/pull/191973
Will need to rebase once those are merged. 

see commits https://github.com/pytorch/pytorch/pull/192097/changes/c4e0c5c1ef85bb93cd0d6e3cc46cff6446222db7..cb760bb831420502c80f49fa7881c6050a029807 for new edits in this PR
 
## Test plan
- [x] `python -m pytest test/cpp_extensions/test_libtorch_agnostic.py -k 'test_my_index_select or test_my_floor_divide or test_my_is_pinned' -v`

cc @janeyx99

### CODE REVIEW

File: test/cpp_extensions/test_libtorch_agnostic.py [hunk 1]
```diff
@@ -313,6 +313,46 @@ def test_my_view_dtype(self, device):
...
+
+    @onlyCPU
+    @skipIfTorchVersionLessThan(2, 10)
+    def test_my_is_pinned(self, device):
+        import libtorch_agn_2_10 as libtorch_agnostic
+
+        t = torch.randn(2, 3, device=device)
+        self.assertFalse(libtorch_agnostic.ops.my_is_pinned(t))
+
+        if not torch.cuda.is_available():
```
Comment (janeyx99) [inline test/cpp_extensions/test_libtorch_agnostic.py]: this should be extraneous

File: test/cpp_extensions/test_libtorch_agnostic.py [hunk 2]
```diff
@@ -313,6 +313,46 @@ def test_my_view_dtype(self, device):
...
+        index = torch.tensor([0, 2], device=device)
+        result = libtorch_agnostic.ops.my_index_select(t, 0, index)
+        self.assertEqual(result, torch.index_select(t, 0, index))
+
+    @skipIfTorchVersionLessThan(2, 10)
+    def test_my_floor_divide(self, device):
+        import libtorch_agn_2_10 as libtorch_agnostic
+
+        a = torch.randint(1, 20, (3, 4), device=device, dtype=torch.int64)
+        b = torch.randint(1, 5, (3, 4), device=device, dtype=torch.int64)
```
Comment (janeyx99) [inline test/cpp_extensions/test_libtorch_agnostic.py:352]: use negative divisors to actually execute floor_divide well

Comment (cleonard530) [reply test/cpp_extensions/test_libtorch_agnostic.py:352]: Done, and it's probably not necessary, but I also added negative values to the numerator range as well (`[-19, 19]` insteadl of `[1, 19]`)

File: test/cpp_extensions/test_libtorch_agnostic.py [hunk 3]
```diff
@@ -335,6 +335,50 @@ def test_my_view_dtype(self, device):
         self.assertEqual(result, expected)
         self.assertEqual(result.data_ptr(), t.data_ptr())
 
+    @skipIfTorchVersionLessThan(2, 10)
+    def test_my_index_select(self, device):
+        import libtorch_agn_2_10 as libtorch_agnostic
+
+        t = torch.randn(3, 4, device=device)
+        index = torch.tensor([0, 2], device=device)
+        result = libtorch_agnostic.ops.my_index_select(t, 0, index)
```
Comment (janeyx99) [inline test/cpp_extensions/test_libtorch_agnostic.py]: try a negative dim here

File: test/cpp_extensions/libtorch_agn_2_10_extension/csrc/my_is_pinned.cpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#include <torch/csrc/stable/library.h>
+#include <torch/csrc/stable/ops.h>
+#include <torch/csrc/stable/tensor.h>
+
+using torch::stable::Tensor;
+
+bool my_is_pinned(Tensor self) {
```
Comment (janeyx99) [inline test/cpp_extensions/libtorch_agn_2_10_extension/csrc/my_is_pinned.cpp]: not const Tensor&?

let's be consistent with the signatures

File: test/cpp_extensions/test_libtorch_agnostic.py [hunk 5]
```diff
@@ -335,6 +335,50 @@ def test_my_view_dtype(self, device):
...
+            torch.floor_divide(a, b_1d),
+        )
+        self.assertEqual(
+            libtorch_agnostic.ops.my_floor_divide(a, -b_1d),
+            torch.floor_divide(a, -b_1d),
+        )
+
+    @onlyCPU
+    @skipIfTorchVersionLessThan(2, 10)
+    def test_my_is_pinned(self, device):
```
Comment (janeyx99) [inline test/cpp_extensions/test_libtorch_agnostic.py]: this will probs need two tests, the first being cpu only and the second needing an accelerator.

**Review by janeyx99 — APPROVED** [review]
Approving trusting that you'll address all my review comments
